### DISCUSSION TITLE
PrismaClientInitializationError

### DISCUSSION THREAD

**Discussion by HenryCBQ** [prisma/prisma#23426] [Q&A] [answered]
## When I try to make any query to the DB (PostgreSQL) I get the following error:**

⨯ PrismaClientInitializationError:
Invalid `prisma.user.findMany()` invocation:
Can't reach database server at `127.0.0.1`:`5432`
Please make sure your database server is running at `127.0.0.1`:`5432`.
 {
  clientVersion: '5.10.2',
  errorCode: undefined
}

It seems like a connection error, but I have already checked the environment variable and it is correct, the migration has already been run successfully and also the generation (npx prisma generate). I also made sure that the environment variable is read correctly and even did connection tests with a prisma validate. 

I hope you can help me, I have not been able to find the cause of the error.

### Information about Prisma Schema, Client Queries and Environment (optional)

-schema.prisma
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url = env("DATABASE_URL")
}

model User {
  id Int @id @default(autoincrement())
  email String @unique
  username String @unique
  password String
  createdAt DateTime @default(now())
  updateAt DateTime @updatedAt
}
```

-Prisma Client querie
```
import { NextResponse } from 'next/server'
import prisma from '@/libs/db'

export async function POST(request){
    const data = await request.json()
    const newUser = await prisma.user.create({
        data
    })
    return NextResponse.json(newUser)
}
```

-db.js:
```
import { PrismaClient } from '@prisma/client'
const prismaClientSingleton = () => {
    return new PrismaClient({log: ['query', 'info', 'warn', 'error']})
}
const globalForPrisma = globalThis;
const prisma = globalForPrisma.prisma ?? prismaClientSingleton()
export default prisma
if(process.env.NODE_ENV !== 'production') globalForPrisma.prisma = prisma
```

-.env example structure: DATABASE_URL="postgresql://postgres:pass@127.0.0.1:5432/nextjs_prisma_auth?schema=public"

- OS: Windows 11
- Database: PostgreSQL
- Node.js version: v20.10.0
- Prisma Cliente version: 5.10.2
- Nextjs version: 14.1.2

**Comment by ludralph** [comment, marked as answer]
Hi @HenryCBQ 👋 

Thank you for raising this question.

Can you reconfirm that your PostgreSQL server is running and accessible at `127.0.0.1:5432`. You can verify this by trying to connect to your database using a database client like [pgAdmin](https://www.pgadmin.org/) or [TablePlus](https://tableplus.com/) or the psql command-line utility. Also, if you have special characters in your password for example `@`, you will need to percent encode it as shown in this section of the [documentation](https://www.prisma.io/docs/orm/reference/connection-urls#special-characters)

**Reply by HenryCBQ** [reply]
Hi Ludralph, indeed, I confirm that the password has special characters and these characters must be encoded in the connection URL. I tested it and it works correctly now, thank you very much I hope this helps other developers. Greetings.

**Reply by zainasr** [reply]
I HAVE BEEN facing this issue since last week. I tried everything but there is some special chracter in my password that causes this issue.
Now its properly working. Thankyou so much man for your guide.
